tests/unit/lib/Any_unittest.cc

```diff
@@ -96,6 +96,29 @@ class Object : public CSGObject
 	}
 };
 
+class WithFromString
+{
+public:
+	WithFromString() : value()
+	{
+	}
+	WithFromString(const std::string& v) : value(v)
+	{
+	}
+
+	static WithFromString from_string(const std::string& str)
```

Comment:
Do we need to implement this for every enum type? :scream: ?

Reply:
Yes. We should use some fancy enum like https://github.com/aantron/better-enums you mentioned.

This is the only scalable way I can think of.